rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -3003,6 +3007,40 @@ public void setNeededNodes(int neededNodes) {
         this.monitoring.setNeededNodes(neededNodes);
     }
 
+    @Override
+    public Map<String, List<String>> getInfrasToPoliciesMapping() {
+        Map<String, List<String>> mapping = new HashMap<>();
+        String fileName = null;
+        try {
+            fileName = PAResourceManagerProperties.RM_NODESOURCE_INFRA_POLICY_MAPPING.getValueAsString();
+            if (!(new File(fileName).isAbsolute())) {
+                // file path is relative, so we complete the path with the prefix RM_Home constant
+                fileName = PAResourceManagerProperties.RM_HOME.getValueAsString() + File.separator + fileName;
+            }
+
+            try (FileReader fr = new FileReader(fileName); BufferedReader br = new BufferedReader(fr)) {
```

Comment:
Optionally, `Files.readAllLines()` provides a smoother way to read the definition file: I bet you can use its output directly as a stream.

Reply:
Grazie mille